macros/filter_exceptions.sql

```diff
@@ -1,21 +1,31 @@
-{% macro filter_exceptions(model_ref) -%}
-  {{ return(adapter.dispatch('filter_exceptions', 'dbt_project_evaluator')(model_ref)) }}
+{% macro filter_exceptions(model_ref, model_query) -%}
+  {{ return(adapter.dispatch('filter_exceptions', 'dbt_project_evaluator')(model_ref, model_query)) }}
 {%- endmacro %}
 
-{% macro default__filter_exceptions(model_ref) %}
+{% macro default__filter_exceptions(model_ref, model_query) %}
 
 {% set query_filters %}
 select
     column_name,
     id_to_exclude
 from {{ ref('dbt_project_evaluator_exceptions') }}
-where fct_name = '{{ model_ref.name }}'
+where '{{ model_ref.name }}' like fct_name
 {% endset %}
 
 {% if execute and flags.WHICH not in ['compile'] %}
+
+    {% set all_columns = get_columns_in_query(model_query) %}
+
     where 1 = 1
     {% for row_filter in run_query(query_filters) %}
-        and {{ row_filter[0] }} not like '{{ row_filter[1] }}'
+        {% if '%' in row_filter[0] %}
+            {% set column_pattern = row_filter[0] | replace('%', '.*') %}
+            {% for column_name in all_columns if modules.re.match(column_pattern, column_name) %}
```

Comment:
I don't think that we can move so easily between the `LIKE` statement and regexes.

For example, `_` is also a character that is allowed in `LIKE` statements and translate to `.` in regex but would not work in that case.